src/psyclone/domain/lfric/transformations/lfric_alg_invoke_2_psy_call_trans.py

```diff
@@ -58,16 +67,323 @@ def validate(self, node, options=None):
           :py:class:`psyclone.domain.lfric.algorithm.LFRicAlgorithmInvokeCall`
         :param options: a dictionary with options for transformations.
         :type options: Optional[Dict[str, Any]]
+        :param options["kernels"]: this option provides a list of \
+            LFRic kernels for this LFRic Algorithm Invoke call.
+        :type options["kernels"]: \
+            List[:py:class:`psyclone.psyir.nodes.Container`]
 
         :raises TransformationError: if the supplied call argument is \
             not a PSyIR AlgorithmInvokeCall node.
+        :raises TransformationError: if kernels options are not provided.
 
         '''
         if not isinstance(node, LFRicAlgorithmInvokeCall):
             raise TransformationError(
                 f"Error in {self.name} transformation. The supplied call "
                 f"argument should be an `LFRicAlgorithmInvokeCall` node but "
                 f"found '{type(node).__name__}'.")
+        try:
+            kernels = options["kernels"]
+        except (KeyError, TypeError) as info:
+            raise TransformationError(
+                "Kernels metadata must be passed into the "
+                "LFRicAlgInvoke2PSyCallTrans transformation but this was not "
+                "found.") from info
+        if not isinstance(kernels, dict):
+            raise TransformationError(
+                f"The value of 'kernels' in the options argument must be a "
+                f"dictionary but found '{type(kernels).__name__}'.")
+        for kern_call in node.children:
+            if isinstance(kern_call, LFRicBuiltinFunctor):
+                # Skip builtins as their metadata is stored internally
+                continue
+            try:
+                _ = kernels[id(kern_call)]
```

Comment:
We have to use `id` because a `KernelFunctor` and an `AlgorithmInvokeCall` are not hashable. We can't use the name of the functor because we can have more than one functor with the same name within an invoke.